### PR TITLE
RPM dependencies for foreman_google package

### PR DESCRIPTION
**Pull request by stejskalleos**
Dependencies for https://github.com/theforeman/foreman-packaging/pull/7777

### CODE REVIEW

**Comment by stejskalleos** [comment]
[test rpm]

**Comment by stejskalleos** [comment]
[test rpm]

**Comment by stejskalleos** [comment]
@evgeni Looks like there is a problem with `TASK [build_package : Release to koji]` for `rubygem-google-protobuf`:
```
TASK [build_package : Set tito_releasers] **************************************
 ok: [rubygem-google-protobuf]
 
 TASK [build_package : Release to koji] *****************************************
 fatal: [rubygem-google-protobuf]: FAILED! => changed=false 
   code: 1
   command:
   - tito
   - release
   - --yes
   - --scratch
   - koji-foreman
   directory: /home/jenkins/workspace/foreman-packaging-rpm-pr-test/packages/foreman//rubygem-google-protobuf
   msg: Failed to tito release
   output: |-
     ERROR: Error running command: git-annex get
   
     Status code: 256
   
     Command output: get google-protobuf-3.20.1-x86_64-linux.gem (not available)
       No other repository is known to contain the file.
   
       (Note that these git remotes have annex-ignore set: origin)
     failed
     git-annex: get: 1 failed
   
     Creating output directory: /tmp/tito
     Will release to the following targets: koji-foreman
     Releasing to target: koji-foreman
     Working in: /tmp/tito/release-rubygem-google-protobufb_k4um
     Building release in Koji...
     WARNING: rubygem-google-protobuf not specified in whitelist for foreman-nightly-nonscl-rhel7
        Package *NOT* submitted to Koji.
     Traceback (most recent call last):
       File "/usr/bin/tito", line 23, in <module>
         CLI().main(sys.argv[1:])
       File "/usr/lib/python2.7/site-packages/tito/cli.py", line 203, in main
         return module.main(argv)
       File "/usr/lib/python2.7/site-packages/tito/cli.py", line 598, in main
         scratch=self.options.scratch)
       File "/usr/lib/python2.7/site-packages/tito/release/main.py", line 463, in release
         self._koji_release()
       File "/usr/lib/python2.7/site-packages/tito/release/main.py", line 528, in _koji_release
         builder.srpm(dist=disttag)
       File "/usr/lib/python2.7/site-packages/tito/builder/main.py", line 206, in srpm
         self.tgz()
       File "/usr/lib/python2.7/site-packages/tito/builder/main.py", line 612, in tgz
         self._setup_sources()
       File "/home/jenkins/workspace/foreman-packaging-rpm-pr-test/obal/obal/tito_extensions/git_annex_spec_builder.py", line 161, in _setup_sources
         run_command("git-annex get")
       File "/usr/lib/python2.7/site-packages/tito/common.py", line 426, in run_command
         raise RunCommandException(command, status, output)
     tito.exception.RunCommandException: Error running command: git-annex get
```
I added the gem same way as the others, no changes from my side, any ideas what's wrong?

**Comment by ehelms** [comment]
I am reminded of https://github.com/theforeman/foreman-packaging/pull/7438 which would imply that we need to wait to bring in these changes until we drop EL7 (which is happening in the next couple of weeks).

**Comment by ehelms** [comment]
@stejskalleos if you rebase this, you will now find that it only does the EL8 builds

File: packages/foreman/rubygem-faraday/rubygem-faraday.spec [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
 %{?scl:%scl_package rubygem-%{gem_name}}
 %{!?scl:%global pkg_name %{name}}
 
 %global gem_name faraday
+%global gem_require_name %{gem_name}
 
 Name: %{?scl_prefix}rubygem-%{gem_name}
-Version: 0.17.3
-Release: 2%{?dist}
+Version: 2.3.0
```
[the remark this answers is not in the document]
Comment (stejskalleos) [reply packages/foreman/rubygem-faraday/rubygem-faraday.spec:9]: Alright, so what we can do?

File: packages/foreman/rubygem-googleauth/rubygem-googleauth.spec [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
 Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
 
 # start specfile generated dependencies
 Requires: %{?scl_prefix_ruby}ruby(release)
-Requires: %{?scl_prefix_ruby}ruby >= 2.4.0
+Requires: %{?scl_prefix_ruby}ruby >= 2.5
 Requires: %{?scl_prefix_ruby}ruby(rubygems)
 Requires: %{?scl_prefix}rubygem(faraday) >= 0.17.3
-Requires: %{?scl_prefix}rubygem(faraday) < 2.0
+Requires: %{?scl_prefix}rubygem(faraday) < 3.a
```
Comment (ekohl) [inline packages/foreman/rubygem-googleauth/rubygem-googleauth.spec:22]: I don't think RPM supports this version number.

Comment (stejskalleos) [reply packages/foreman/rubygem-googleauth/rubygem-googleauth.spec:22]: So should I reverted back to `2.0`?

**Comment by ekohl** [comment]
We've made a lot of changes (added various google gems, rewritten our gem templates). At this point it may be easier to do this PR again.

**Pull request closed without merging**